src/modules/m_classban.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+ * InspIRCd -- Internet Relay Chat Daemon
+ *
+ *   Copyright (C) 2016 Johanna Abrahamsson <johanna-a@users.noreply.github.com>
+ *
+ * This file is part of InspIRCd.  InspIRCd is free software: you can
+ * redistribute it and/or modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation, version 2.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+#include "inspircd.h"
+
+class ModuleClassBan : public Module
+{
+ public:
+	Version GetVersion() CXX11_OVERRIDE
+	{
+		return Version("Class 'n' - Connection class ban", VF_VENDOR | VF_OPTCOMMON);
+	}
+
+	ModuleClassBan() {};
+
+	ModResult OnCheckBan(User* user, Channel* c, const std::string& mask) CXX11_OVERRIDE
+	{
+		LocalUser* localUser = IS_LOCAL(user);
+		if (localUser && (mask.length() > 2) && (mask[0] == 'n') && (mask[1] == ':'))
```

Comment:
Put the first check (localuser) in parenthesis as well to avoid inconsistency with the other checks in this if